From ac91387e38c2d12013331e13baf0705d178197ba Mon Sep 17 00:00:00 2001
From: Doug Ledford <dledford@redhat.com>
Date: Wed, 24 Sep 2014 15:21:39 -0400
Subject: [infiniband] ocrdma: Fix load time panic during GID table init

Message-id: <e9aa15e8294f9b3f297345f28d50d28bf332bf86.1411571481.git.dledford@redhat.com>
Patchwork-id: 96636
O-Subject: [Patch RHEL7 04/48] RDMA/ocrdma: Fix load time panic during GID table init
Bugzilla: 1081067 1088577
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Devesh Sharma <devesh.sharma@emulex.com>

We should use rdma_vlan_dev_real_dev() instead of using vlan_dev_real_dev()
when building the GID table for a vlan interface.

Signed-off-by: Devesh Sharma <devesh.sharma@emulex.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
(cherry picked from commit 09de3f1313a30d8a22e488c9a5b96a9560cae96d)
Signed-off-by: Doug Ledford <dledford@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/infiniband/hw/ocrdma/ocrdma_main.c b/drivers/infiniband/hw/ocrdma/ocrdma_main.c
index 2ca86ca..1a8a945 100644
--- a/drivers/infiniband/hw/ocrdma/ocrdma_main.c
+++ b/drivers/infiniband/hw/ocrdma/ocrdma_main.c
@@ -127,7 +127,7 @@ static int ocrdma_addr_event(unsigned long event, struct net_device *netdev,
 
  is_vlan = netdev->priv_flags & IFF_802_1Q_VLAN;
  if (is_vlan)
-  netdev = vlan_dev_real_dev(netdev);
+  netdev = rdma_vlan_dev_real_dev(netdev);
 
  rcu_read_lock();
  list_for_each_entry_rcu(dev, &ocrdma_dev_list, entry) {
-- 
1.7.1